packages/integration-karma/test/light-dom/restriction/index.spec.js

```diff
@@ -0,0 +1,31 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+
+import Component from 'x/component';
+
+describe('restriction', () => {
+    beforeEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+    });
+
+    afterEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    });
+    it('should not restrict setting innerHTML on non-portaled light DOM component', () => {
+        const elm = createElement('x-component', { is: Component });
+        document.body.appendChild(elm);
+
+        expect(() => {
+            elm.querySelector('div').innerHTML = '<span>hello</span>';
+        }).not.toLogErrorDev();
+        expect(elm.querySelector('div span').textContent).toEqual('hello');
+    });
+
+    it('should restrict setting outerHTML on light DOM component', () => {
+        const elm = createElement('x-component', { is: Component });
+        document.body.appendChild(elm);
+
+        expect(() => {
+            elm.querySelector('div').outerHTML = 'foo';
+        }).toThrowError(TypeError, 'Invalid attempt to set outerHTML on Element.');
+    });
```

Comment:
`patchElementWithRestrictions` also restricts setting `outerHTML`. I don't know what purpose this serves, but it seems unrelated to portals.

Reply:
I also don't think it is related to portals. The goal is to avoid that components get unounted using `outHTML`. That said it is quite an edge case.